src/lib/services/api-token-service.ts

```diff
@@ -79,6 +96,13 @@ export class ApiTokenService {
         }
     }
 
+    async updateLastSeen(): Promise<void> {
+        if (this.lastSeenSecrets.length > 0) {
+            await this.store.markSeenAt(this.lastSeenSecrets);
+            this.lastSeenSecrets = [];
```

Comment:
We could lose some last seen in between due to the interleaving of different operations. I don't know if there's an atomic way of doing this... you may know better, but I googled this:
```typescript
let toStore = [];
[this.lastSeenSecrets, toStore] = [toStore, this.lastSeenSecrets];
await this.store.markSeenAt(toStore);
```

Reply:
Interesting! That could work, thanks 👍